Framework/DataObjects/inc/MantidDataObjects/PeaksWorkspace.h

```diff
@@ -94,6 +94,8 @@ class DLLExport PeaksWorkspace : public Mantid::API::IPeaksWorkspace {
   std::string getConvention() const override;
   void removePeak(int peakNum) override;
   void addPeak(const Geometry::IPeak &ipeak) override;
+  /// Move a peak object into this peaks workspace
+  void addPeak(Peak &&ipeak);
```

Comment:
If you make other changes, change the variable name here and in Peaksworkspace.cpp to `peak`.

Beyond that, I'll defer to the other reviews.